runtime/src/bank/fee_distribution.rs

```diff
@@ -230,20 +236,29 @@ impl Bank {
                             ));
                         }
                         Err(err) => {
+                            debug!(
+                                "Burned {} lamport rent fee instead of sending to {} due to {}",
+                                rent_to_be_paid, pubkey, err
+                            );
+
                             // overflow adding lamports or resulting account is invalid
-                            self.capitalization.fetch_sub(rent_to_be_paid, Relaxed);
-                            // notify validator that their fees were burned
-                            if self.collector_id == pubkey {
-                                error!(
-                                    "Burned {rent_to_be_paid} lamport rent fee instead of sending to {pubkey} due to {err}",
-                                );
-                            }
+                            // so burn lamports and track lamports burned per slot
+                            rent_to_burn = rent_to_burn.saturating_add(rent_to_be_paid);
                         }
                     }
                 }
             });
         self.rewards.write().unwrap().append(&mut rewards);
 
+        if rent_to_burn > 0 {
+            self.capitalization.fetch_sub(rent_to_burn, Relaxed);
```

Comment:
This change to a single subtraction per loop looks fine, but since it was potentially substantive, it would have been nice if called out in its own commit.